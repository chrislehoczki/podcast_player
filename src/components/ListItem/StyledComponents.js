import styled from "styled-components";
import theme from "../../styles/theme";

export const ListContainer = styled.div`
  display: flex;
  padding: 10px;
  border: 1px solid ${theme.primary};
`;

export const DetailContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
  width: 400px;
`;

export const MainDetail = styled.h4`
  margin: 0;
`;

export const SubDetail = styled.div`
  margin: 0;
  font-size: 14px;
`;

export const Duration = styled.p`
  margin: 3px 0;
  font-size: 14px;
`;
