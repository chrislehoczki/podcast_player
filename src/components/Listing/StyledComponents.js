import styled from "styled-components";
import theme from "../../styles/theme";

export const Container = styled.div`
  width: 100%;
  height: calc(100vh - 40px - 400px);
  overflow-y: auto;
  ::-webkit-scrollbar {
    width: 10px;
  }
  ::-webkit-scrollbar-track {
    background: 0;
  }
  ::-webkit-scrollbar-thumb {
    background: ${theme.primary};
  }
  @media only screen and (max-width: 400px) {
    height: calc(100vh - 400px);
  }
`;
