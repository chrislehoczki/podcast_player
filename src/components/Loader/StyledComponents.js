import styled from "styled-components";

export const Ring = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  &:after {
    content: " ";
    display: block;
    width: 55%;
    height: 55%;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const LoaderContainer = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;
