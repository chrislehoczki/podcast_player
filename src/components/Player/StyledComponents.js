import styled from "styled-components";
import theme from "../../styles/theme";

export const PlayerContainer = styled.div`
  width: 100%;
  height: 400px;
  position: relative;
  background: ${theme.background};
  border-bottom: 2px solid ${theme.primary};
`;

export const Image = styled.div`
  width: 150px;
  height: 150px;
  min-height: 150px;
  background: url(${props => props.img}) no-repeat center center;
  background-size: cover;
`;

export const Details = styled.div`
  height: 60%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  padding: 20px 10px;
`;

export const Title = styled.h1`
  margin: 0;
  font-size: 18px;
  margin: 20px 0 5px 0;
  line-height: 22px;
`;

export const Author = styled.h2`
  font-size: 16px;
  margin: 0;
`;

export const Controls = styled.div`
  display: flex;
  vertical-align: center;
  align-items: center;
  width: 100%;
  height: 20%;
  justify-content: center;
`;
