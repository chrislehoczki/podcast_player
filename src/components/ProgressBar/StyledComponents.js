import styled from "styled-components";
import theme from "../../styles/theme";

const sliderThumbStyles = props => `
  width: 15px;
  height: 15px;
  border-radius: 100%;
  background: ${theme.primary};
`;

export const Container = styled.div`
  position: absolute;
  bottom: 25px;
  width: 80%;
  margin: 0;
  display: flex;
  align-items: center;
  color: ${theme.primary};
  .slider {
    flex: 6;
    -webkit-appearance: none;
    width: 100%;
    height: 5px;
    border-radius: 5px;
    background: ${theme.primary};
    outline: none;
    &::-webkit-slider-thumb {
      -webkit-appearance: none;
      appearance: none;
      ${props => sliderThumbStyles(props)}
    }
    &::-moz-range-thumb {
      ${props => sliderThumbStyles(props)}
    }
  }
`;
