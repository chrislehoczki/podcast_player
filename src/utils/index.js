const humanReadableTime = duration => {
  const seconds = Math.floor((duration / 1000) % 60);
  return {
    minutes: Math.floor(duration / 1000 / 60),
    seconds: seconds < 10 ? `0${seconds}` : seconds
  };
};

const next = (list, currentId) => {
  const index = list.findIndex(item => item.id === currentId);
  if (index === list.length - 1) {
    return list[0];
  }
  return list[index + 1];
};
const prev = (list, currentId) => {
  const index = list.findIndex(item => item.id === currentId);
  if (index === 0) {
    return list[list.length - 1];
  }
  return list[index - 1];
};

export { humanReadableTime, next, prev };
