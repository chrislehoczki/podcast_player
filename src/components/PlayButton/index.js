import React from "react";
import PropTypes from "prop-types";
import { Button } from "../../styles/theme";
import Loader from "../Loader";

const PlayButton = ({
  loaded,
  playing,
  stopPodcast,
  playPodcast,
  podcastId
}) => {
  return !loaded ? (
    <Loader size={60} />
  ) : playing ? (
    <Button
      img={"icons/pause.png"}
      size={60}
      border={true}
      onClick={() => loaded && stopPodcast()}
    />
  ) : (
    <Button
      img={"icons/play.png"}
      size={60}
      border={true}
      onClick={() => playPodcast({ id: podcastId })}
    />
  );
};

PlayButton.propTypes = {
  playPodcast: PropTypes.func,
  stopPodcast: PropTypes.func,
  loaded: PropTypes.bool,
  playing: PropTypes.bool,
  podcastId: PropTypes.string
};

export default PlayButton;
