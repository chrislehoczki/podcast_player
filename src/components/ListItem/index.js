import React from "react";
import PropTypes from "prop-types";
import { humanReadableTime } from "../../utils";

import {
  ListContainer,
  DetailContainer,
  MainDetail,
  SubDetail,
  Duration
} from "./StyledComponents";
import PlayButton from "../PlayButton";
const ListItem = ({
  title,
  description,
  duration,
  playPodcast,
  stopPodcast,
  playing,
  loaded
}) => {
  const { minutes, seconds } = humanReadableTime(duration);
  return (
    <ListContainer>
      <PlayButton
        stopPodcast={stopPodcast}
        playPodcast={playPodcast}
        loaded={loaded}
        playing={playing}
      />
      <DetailContainer>
        <MainDetail>{title}</MainDetail>
        <Duration>{`${minutes}:${seconds}`}</Duration>
        <SubDetail dangerouslySetInnerHTML={{ __html: description }} />
      </DetailContainer>
    </ListContainer>
  );
};

ListItem.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  duration: PropTypes.number,
  playPodcast: PropTypes.func,
  stopPodcast: PropTypes.func,
  playing: PropTypes.bool,
  loaded: PropTypes.bool
};

export default ListItem;
