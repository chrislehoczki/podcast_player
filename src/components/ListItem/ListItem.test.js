import React from "react";
import { shallow } from "enzyme";
import ListItem from "./index";
import mock from "../../test/mock";
import { humanReadableTime } from "../../utils";
import { MainDetail, SubDetail, Duration } from "./StyledComponents";
import PlayButton from "../PlayButton";
const { title, description, duration } = mock[0];

describe("<ListItem />", () => {
  it("Renders correctly", () => {
    const component = shallow(
      <ListItem
        title={title}
        description={description}
        duration={duration}
        playing={false}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("Renders data fields correctly", () => {
    const component = shallow(
      <ListItem title={title} description={description} duration={duration} />
    );
    const { minutes, seconds } = humanReadableTime(duration);

    expect(component.find(MainDetail).text()).toEqual(title);
    expect(component.find(Duration).text()).toEqual(`${minutes}:${seconds}`);
    expect(component.find(SubDetail).html()).toContain(description);
    expect(component).toMatchSnapshot();
  });
  it("Renders a <PlayButton /> component", () => {
    const component = shallow(
      <ListItem title={title} description={description} duration={duration} />
    );
    expect(component.find(PlayButton)).toExist();
  });
});
