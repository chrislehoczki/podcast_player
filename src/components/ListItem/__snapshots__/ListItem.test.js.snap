// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Renders correctly 1`] = `
<styled.div>
  <PlayButton
    playing={false}
  />
  <styled.div>
    <styled.h4>
      Nick Horrocks: Mastering the ABC’s of Startup Funding
    </styled.h4>
    <styled.p>
      25:13
    </styled.p>
    <styled.div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Recently I had the pleasure of sitting down with <b>Nick Horrocks</b> founder of X.0 Advisory — a new startup corporate finance boutique.</p>

<p>We discuss everything from how to raise money as a start up through to starting a business alone with only your cat for company.</p>",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ListItem /> Renders data fields correctly 1`] = `
<styled.div>
  <PlayButton />
  <styled.div>
    <styled.h4>
      Nick Horrocks: Mastering the ABC’s of Startup Funding
    </styled.h4>
    <styled.p>
      25:13
    </styled.p>
    <styled.div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Recently I had the pleasure of sitting down with <b>Nick Horrocks</b> founder of X.0 Advisory — a new startup corporate finance boutique.</p>

<p>We discuss everything from how to raise money as a start up through to starting a business alone with only your cat for company.</p>",
        }
      }
    />
  </styled.div>
</styled.div>
`;
